File: CartagoFrontV2-master/src/app/services/referenciel/group.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Group } from '../../models/group';

@Injectable({
  providedIn: 'root',
})
export class GroupService {
  constructor(private http: HttpClient) {}

  getAllGroups(): Observable<Group[]> {
    return this.http.get<Group[]>('http://localhost:8080/Groups');
  }

  addGroup(group: Group): Observable<Group> {
    return this.http.post<Group>('http://localhost:8080/Groups/', group);
  }

  updateGroup(group: Group): Observable<Group> {
    return this.http.put<Group>('http://localhost:8080/Groups/', group);
  }

  deleteGroup(id: number): Observable<Group> {
    return this.http.delete<Group>('http://localhost:8080/Groups/' + id);
  }
}

File: MicroServices/MicroServices/referenciel-service/src/main/java/com/bfi/referentiel/service/GroupService.java
package com.bfi.referentiel.service;

import com.bfi.referentiel.common.dto.GroupDTO;

import java.util.List;

public interface GroupService {

    void addGroup(GroupDTO group);

    void updateGroup(GroupDTO group);

    void deleteGroup(Long id);

    List<GroupDTO> getAllGroups();
}
